servers/src/test/java/tachyon/worker/block/evictor/EvictorTest.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.evictor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Files;
+
+import tachyon.Constants;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+
+/**
+ * This class tests general behavior of all types of {@link Evictor} defined in {@link EvictorType}.
+ */
+@RunWith(Parameterized.class)
+public class EvictorTest {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static final int USER_ID = 2;
+  private static final long BLOCK_ID = 10;
+  private BlockMetadataManager mMeta;
+  private EvictorType mEvictorType;
+  private Evictor mEvictor;
+  private File mTempFolder;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    // Run this test against all types of Evictors
+    List<Object[]> list = new ArrayList<Object[]>();
+    for (EvictorType type : EvictorType.values()) {
+      if (type != EvictorType.DEFAULT) {
+        list.add(new Object[] {type});
+      }
+    }
+    return list;
+  }
+
+  public EvictorTest(EvictorType evictorType) {
+    mEvictorType = evictorType;
+  }
+
+  @Before
+  public final void before() throws IOException {
+    mTempFolder = Files.createTempDir();
+    mMeta = EvictorTestUtils.defaultMetadataManager(mTempFolder.getAbsolutePath());
+    mEvictor = EvictorFactory.create(mEvictorType, mMeta);
+  }
+
+  @After
+  public final void after() throws IOException {
+    FileUtils.forceDelete(mTempFolder);
+  }
+
+  @Test
+  public void noNeedToEvictTest1() throws IOException {
+    // metadata manager is just created, no cached block in Evictor
```

Comment:
NIT: I really appreciate the description of the test setting in each test case!! One trick to make the test description even better is to also describe what we expect from the test. For example in this test case, add in the above comment that, we expect the proposed plan is empty. :)